### PR TITLE
Meson fix for `libdecaf` detection

### PR DESCRIPTION
**Pull request by fredmorcos**
### Short description
This fixes `libdecaf` detection on Github runners. The library is installed in `/usr/local` and, when given custom directories to search in, meson doesn't add the `/lib` or `/include` postfix so we need to add those ourselves.

When given `/usr/local`, strace shows that meson searches for the library files in the given directories verbatim:
```
newfstatat(AT_FDCWD, "/usr/local/libdecaf.so", 0x7ffd8d377be0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/usr/local/decaf.so", 0x7ffd8d377be0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/usr/local/libdecaf.a", 0x7ffd8d377be0, 0) = -1 ENOENT (No such file or directory)
```

There's a tiny cleanup for the postgres backend.

*Do not merge until we've tested it again on the runner*.

### Checklist
I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: meson/libdecaf/meson.build [hunk 1]
```diff
@@ -6,8 +6,18 @@ header_path = false
...
   dep_libdecaf = dependency('libdecaf', required: false)
+
   if not dep_libdecaf.found()
-    dep_libdecaf = cxx.find_library('decaf', dirs: ['/usr', '/usr' / 'local'], required: opt_libdecaf)
+    dep_libdecaf = cxx.find_library(
+      'decaf',
+      dirs: [
+        '/usr',
+        '/usr' / 'lib',
+        '/usr' / 'local',
```
Comment (Habbie) [inline meson/libdecaf/meson.build:16]: why keep the versions without /lib suffix?

**Review by Habbie — APPROVED** [review]
approved with a question

**Comment by fredmorcos** [comment]
I tested this now on the runner and on my local machine and it works.

@Habbie Could you please test on different machines (if you have any)?
@romeroalx Could you please test (again) on the runner to make sure?
@omoerbeek Could you please test this on Open/FreeBSD and MacOs?

I'm keeping the other commits around for "historical reasons". The relevant commits to review are those:
- https://github.com/PowerDNS/pdns/pull/14031/commits/a3fb0fb2e2d00747458c6bf8e1243d733963ce88
- https://github.com/PowerDNS/pdns/pull/14031/commits/78f9b7cc9f0b88bc68d43dfe244d39bcf4d9bb92

**Review by romeroalx — APPROVED** [review]

**Comment by Habbie** [comment]
> @Habbie Could you please test on different Linuxes (if you have any)?

Debian 12, with libdecaf via apt (which makes me realise we could do that on the runners too I think!):

master:

```
Run-time dependency libdecaf found: NO (tried pkgconfig and cmake)
Library decaf found: YES
Message: Libdecaf: Checking for /usr/include/decaf.hxx
Has header "decaf.hxx" with dependency -ldecaf: NO 
Message: Libdecaf: Checking for /usr/include/decaf/decaf.hxx
Has header "decaf.hxx" with dependency -ldecaf: YES 
```

but decaf is not actually linked into pdnsutil (and, I bet, pdns_server)

This PR:

```
Run-time dependency libdecaf found: NO (tried pkgconfig and cmake)
Has header "decaf.hxx" : YES 
Has header "decaf/spongerng.hxx" : YES 
Has header "decaf/eddsa.hxx" : YES 
Library decaf found: YES
```

but decaf is not linked into pdnsutil and pdns_server

**Comment by Habbie** [comment]
Hold on. I made a testing mistake.

**Comment by Habbie** [comment]
> Hold on. I made a testing mistake.

previous comment updated. Will now approve the PR again :)

**Review by Habbie — APPROVED** [review]

**Comment by fredmorcos** [comment]
> > @Habbie Could you please test on different Linuxes (if you have any)?
> 
> Debian 12, with libdecaf via apt (which makes me realise we could do that on the runners too I think!):
> 
> master:
> 
> ```
> Run-time dependency libdecaf found: NO (tried pkgconfig and cmake)
> Library decaf found: YES
> Message: Libdecaf: Checking for /usr/include/decaf.hxx
> Has header "decaf.hxx" with dependency -ldecaf: NO 
> Message: Libdecaf: Checking for /usr/include/decaf/decaf.hxx
> Has header "decaf.hxx" with dependency -ldecaf: YES 
> ```
> 
> and decaf works
> 
> This PR:
> 
> ```
> Run-time dependency libdecaf found: NO (tried pkgconfig and cmake)
> Has header "decaf.hxx" : YES 
> Has header "decaf/spongerng.hxx" : YES 
> Has header "decaf/eddsa.hxx" : YES 
> Library decaf found: YES
> ```
> 
> and decaf works

That's pretty much intended. The previous implementation (on master) was a translation from the autotools implementation. This new implementation offloads the search for the library and header file to meson which should be more robust anyway.

**Pull request merged**